winsup/utils/getprocaddr.c

```diff
@@ -231,9 +238,29 @@ main (int argc, char **argv)
       return 1;
     }
 
-  /* This is needed for both thread to inject
-   * and GenerateConsoleCtrlEvent.
-   * Gives 1 sec time. */
-  Sleep (1000);
+  CtrlEvent = CreateEvent (NULL,              // default security attributes
+                           TRUE,              // manual-reset event
+                           FALSE,             // initial state is nonsignaled
+                           TEXT ("CtrlEvent") // object name
+  );
+  if (CtrlEvent == NULL)
+    {
+      output (1, "CreateEvent failed (%d)\n", GetLastError ());
+      return 1;
+    }
+
+  DWORD WaitResult = WaitForSingleObject (CtrlEvent, 10000 /* 10 seconds*/);
```

Comment:
Since the `WaitResult` is used immediately and only once, why not move the call to `WaitForSingleObject()` into the conditional directly?